mypy_django_plugin/transformers/managers.py

```diff
@@ -105,6 +105,8 @@ def get_funcdef_type(definition: Union[FuncBase, Decorator, None]) -> Optional[P
             ret_type = Instance(queryset_info, [manager_instance.args[0], manager_instance.args[0]])
         variables = []
 
+    ret_type = _manage_type_var(ret_type, queryset_info)
```

Comment:
I think your helper methods are doing a bit of duplicated work. While brute forcing a bit, trying to figure out which base class is holding the argument. I'm thinking that since we already have the method name, and with how type vars work we should then _always_ be able to find the type var on the same base that has the method.

We could, instead of _only_ fetching the method on the line below, fetch the base that declares the method, while also honoring the mro.

https://github.com/typeddjango/django-stubs/blob/81a543a070aa9596e3296db88563057f31b1e5df/mypy_django_plugin/transformers/managers.py#L82

So we could adjust to something like:
```python
# Note: Important that we look through bases of bases, according to the mro.
# Make sure to look for method in <base.type.names>
# Note: There's no `get_base_containing_method`, you have to implement that loop.
base_that_has_method: Instance | None = <queryset_info.get_base_containing_method(method_name)>
if base_that_has_method is None:
    return None
method_type = get_funcdef_type(base_that_has_method.type.names[method_name].node)
```

`get_funcdef_type` has to change its signature a little bit, but that's totally fine.

Now, we shouldn't need to dig through any more bases and their arguments. The argument we need _should_ exist on the `base_that_has_method.args` if we can't find it we just `return method_type.ret_type`

Which leads to an additional case for your test having a method with an argument that is not found in the class' args

```python
T = TypeVar("T")
class _MyModelQuerySet(models.QuerySet[_CTE]):
    ...
    def type_var_outside_class(self) -> T: ...
```

`.type_var_outside_class()` should resolve to `T`

Reply:
Thank you for the nice feedback.

I have tried your approach, but in your example https://github.com/typeddjango/django-stubs/pull/1666#discussion_r1311703917, `base_that_has_method.args has _CTE2`. which means that `reveal_type(MyModel.objects.example())` returns `_CTE2`.